### PR TITLE
[add] Get Data

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/general/index.ts [hunk 1]
```diff
@@ -226,3 +226,54 @@ export function downloadFile(downloadConfig: { file: File | null; type: string;
...
+
+    for (const path of newPathArray) {
+      obj = obj[path] as { [key: string]: unknown };
+    }
+
+    return obj ? obj : def;
+
+  } catch (e) {
+    console.error('Error while using getData', e);
+    throw e;
```
Comment (vikaz596) [inline src/general/index.ts:277]: in case of error, should we return default value?

Comment (vyapak1209) [reply src/general/index.ts:277]: Makes sense.. Will fix

File: src/general/index.ts [hunk 2]
```diff
@@ -226,3 +226,54 @@ export function downloadFile(downloadConfig: { file: File | null; type: string;
...
+
+
+/**
+ * Returns the value at given path from the source object. If path is not found then default value is returned.
+ *
+ * @param { { [key: string]: unknown } } obj - Source object
+ * @param {string} path - Path to desired key inside source object
+ *
+ * @remarks
+ * Provide a default value always to avoid unexpected behavior
```
Comment (dheerajsingh123) [inline src/general/index.ts:238]: can be mentioned that this function has same behaviour as lodash getData method

Comment (vikaz596) [reply src/general/index.ts:238]: +1

**Review by dheerajsingh123 — APPROVED** [review]

**Pull request merged**